integrations/servicenow/client.py

```diff
@@ -0,0 +1,103 @@
+from typing import Any, AsyncGenerator
+import base64
+import httpx
+from loguru import logger
+from enum import StrEnum
+from port_ocean.utils import http_async_client
+
+PAGE_SIZE = 100
+
+
+class ObjectKind(StrEnum):
+    GROUP = "sys_user_group"
+    CATALOG = "sc_catalog"
+    INCIDENT = "incident"
+
+
+class ServicenowClient:
+    def __init__(
+        self, servicenow_host: str, servicenow_username: str, servicenow_password: str
+    ):
+        self.servicenow_host = servicenow_host
+        self.servicenow_username = servicenow_username
+        self.servicenow_password = servicenow_password
+        self.http_client = http_async_client
+        self.http_client.headers.update(self.api_auth_params["headers"])
+
+    @property
+    def api_auth_params(self) -> dict[str, Any]:
+        auth_message = f"{self.servicenow_username}:{self.servicenow_password}"
+        auth_bytes = auth_message.encode("ascii")
+        b64_bytes = base64.standard_b64encode(auth_bytes)
+        b64_message = b64_bytes.decode("ascii")
+
+        return {
+            "headers": {
+                "Authorization": f"Basic {b64_message}",
+                "Content-Type": "application/json",
+            },
+        }
+
+    async def get_paginated_resource(
+        self, resource_kind: ObjectKind
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        offset = 0
+
+        while True:
+            params: dict[str, Any] = {
+                "sysparm_offset": offset * PAGE_SIZE,
+                "sysparm_limit": PAGE_SIZE,
+                "sysparm_query": "ORDERBYsys_created_on",
```

Comment:
`"ORDERBYsys_created_on"` typo? 
also why we need it?

Reply:
The results returned by Servicenow is not ordered. And since we are doing pagination, we will need to order the results so the offset and limit can be applied correctly

Reply:
sounds good, my point was that it looks like the `ORDERBY` got joined with `sys_created_on` without any spacing. is that their query language?